hax/hax/motr/__init__.py

```diff
@@ -164,17 +168,19 @@ def send_entrypoint_request_reply(self, message: EntrypointRequest):
                                                  str(process_fid)))
         sess = principal_rm = confds = None
         try:
-            prov = self.consul_util
-            sess = prov.get_leader_session_no_wait()
-            principal_rm = prov.get_session_node(sess)
-            confds = prov.get_confd_list()
+            util = self.consul_util
+            sess = util.get_leader_session_no_wait()
+            principal_rm = util.get_session_node(sess)
+            confds = util.get_confd_list()
+            rm_fid = util.get_rm_fid()
         except Exception:
             LOG.exception('Failed to get the data from Consul.'
                           ' Replying with EAGAIN error code.')
             self._ffi.entrypoint_reply(reply_context, req_id.to_c(), EAGAIN, 0,
                                        make_array(FidStruct, []),
                                        make_array(c.c_char_p, []), 0,
-                                       self.rm_fid.to_c(), None)
+                                       Fid(0, 0).to_c(),
```

Comment:
Why is it zeroed? Is it expected by Motr in case of  rc=EAGAIN ?

Reply:
Its a error case so, either we can set rm_fid to NULL in which case motr anyway sets rm_fid to (0, 0) or we return (0,0) (as in this case).
```
void m0_halon_interface_entrypoint_reply(
                struct m0_halon_interface  *hi,
                const struct m0_uint128    *req_id,
                int                         rc,
                uint32_t                    confd_nr,
                const struct m0_fid        *confd_fid_data,
                const char                **confd_eps_data,
                uint32_t                    confd_quorum,
                const struct m0_fid        *rm_fid,
                const char                 *rm_eps)
{
        struct m0_ha_entrypoint_rep rep;

        M0_ENTRY("hi=%p req_id="U128X_F" rc=%d confd_nr=%"PRIu32" "
                 "confd_quorum=%"PRIu32" rm_fid="FID_F" rm_eps=%s",
                 hi, U128_P(req_id), rc, confd_nr, confd_quorum,
                 FID_P(rm_fid == NULL ? &M0_FID0 : rm_fid), rm_eps);

        if (hi->hif_internal->hii_cfg.hic_log_entrypoint) {
                M0_LOG(M0_WARN, "req_id="U128X_F" rc=%d confd_nr=%"PRIu32" "
                 "confd_quorum=%"PRIu32" rm_fid="FID_F" rm_eps=%s",
                 U128_P(req_id), rc, confd_nr, confd_quorum,
                 FID_P(rm_fid == NULL ? &M0_FID0 : rm_fid), rm_eps);
        }
        rep = (struct m0_ha_entrypoint_rep){
                .hae_quorum        = confd_quorum,
                .hae_confd_fids    = {
                        .af_count = confd_nr,
                        .af_elems = (struct m0_fid *)confd_fid_data,
                },
                .hae_confd_eps     = confd_eps_data,
                .hae_active_rm_fid = rm_fid == NULL ? M0_FID0 : *rm_fid,
                .hae_active_rm_ep  = (char *)rm_eps,
                .hae_control       = rc == 0 ? M0_HA_ENTRYPOINT_CONSUME :
                                               M0_HA_ENTRYPOINT_QUERY,
        };
        m0_ha_entrypoint_reply(&hi->hif_internal->hii_ha, req_id, &rep, NULL);
        M0_LEAVE();
```
As we are returning -EAGAIN, motr will request entrypoint again. So we are good.